tools/gfx/d3d12/render-d3d12.cpp

```diff
@@ -7525,8 +7629,18 @@ Result D3D12Device::ShaderObjectImpl::setResource(ShaderOffset const& offset, IR
     if (offset.bindingRangeIndex >= layout->getBindingRangeCount())
         return SLANG_E_INVALID_ARG;
 
+    ID3D12Device* d3dDevice = static_cast<D3D12Device*>(getDevice())->m_device;
+
     auto& bindingRange = layout->getBindingRange(offset.bindingRangeIndex);
 
+    if (resourceView == nullptr)
+    {
+        // Create null descriptor for the binding.
+        auto destDescriptor = m_descriptorSet.resourceTable.getCpuHandle(
+            bindingRange.baseIndex + (int32_t)offset.bindingArrayIndex);
+        return createNullDescriptor(d3dDevice, destDescriptor, bindingRange);
+    }
```

Comment:
One interesting thing here is that if the user was using `ShaderCursor` to compute the offset they are using, then the `ShaderCursor` would know the Slang type of the parameter it is trying to set, and that same type information would be *exactly* the type information we need in order to create the null descriptor here.

The challenge there is that we probably don't want to require that users always bind via `ShaderCursor`.

I suppose the Right Way to deal with the hypothetical bloat down the road is to ensure that the `BindingType` and `resourceShape` are both stored compactly, rather than burning 32 bits each (or more with padding), but that can way until later (much later).